wandb/sdk/wandb_settings.py

```diff
@@ -43,45 +47,26 @@
         List,
         Optional,
         Union,
+        Tuple,
+        Callable,
+        Set,
     )
 
 logger = logging.getLogger("wandb")
 
-source = (
-    "org",
-    "entity",
-    "project",
-    "system",
-    "workspace",
-    "env",
-    "setup",
-    "settings",
-    "args",
-)
-
-Field = collections.namedtuple("TypedField", ["type", "choices"])
-
-
-defaults = dict(
+defaults: Dict[str, Union[str, int, Tuple]] = dict(
     base_url="https://api.wandb.ai",
     show_warnings=2,
     summary_warnings=5,
-    # old mode field (deprecated in favor of WANDB_OFFLINE=true)
-    _mode=Field(str, ("dryrun", "run", "offline", "online",)),
-    # problem: TODO(jhr): Not implemented yet, needs new name?
-    _problem=Field(str, ("fatal", "warn", "silent",)),
     console="auto",
-    _console=Field(str, ("auto", "redirect", "off", "file", "iowrap", "notebook")),
     git_remote="origin",
-    ignore_globs=[],
-    # anonymous might be set by a config file: "false" and "true"
-    #   or from wandb.init(anonymous=) or environment: "allow", "must", "never"
-    _anonymous=Field(str, ("allow", "must", "never", "false", "true",)),
+    ignore_globs=(),
 )
 
 # env mapping?
-env_prefix = "WANDB_"
-env_settings = dict(
+env_prefix: str = "WANDB_"
+# env_override_suffix: str = "_OVERRIDE"
```

Comment:
Is this not being used?

Reply:
Not in this PR, we do not look at WANDB_PROJECT_OVERRIDE yet. Just putting the code in place to handle it.